### PR TITLE
Implement new API for websub-listener to attach websub-service with custom configurations

### PR DESCRIPTION
**Pull request by ayeshLK**
## Purpose
With this change developers could initialize a `websub:SubscriberService` with custom configurations and attach it to `websub:Listener`.

## User stories
Developer should be able to implement a webhook-service library using `websub:SubscriberService`. When implementing a library developer should be able to pass custom configuration values to underlying `websub:SubscriberService`. But in current implementation `websub:SubscriberService` only support annotation based configurations and hence we could not pass dynamic configuration values to underlying `websub:SubscriberService`. 

## Samples
```ballerina
import ballerina/websub;
import ballerina/log;

service class SimpleWebsubService {
    *websub:SubscriberService;
    remote function onSubscriptionValidationDenied(websub:SubscriptionDeniedError msg) 
                        returns websub:Acknowledgement? {
        log:printInfo("onSubscriptionValidationDenied invoked");
       websub: Acknowledgement ack = {
                  headers: {"header1": "value"},
                  body: {"formparam1": "value1"}
        };
        return ack;
    }

    remote function onSubscriptionVerification(websub:SubscriptionVerification msg)
                        returns websub:SubscriptionVerificationSuccess | websub:SubscriptionVerificationError {
        log:printInfo("onSubscriptionVerification invoked");
        if (msg.hubTopic == "test1") {
            return error websub:SubscriptionVerificationError("Hub topic not supported");
        } else {
            return {};
        }
      }

    remote function onEventNotification(websub:ContentDistributionMessage event) 
                        returns websub:Acknowledgement | websub:SubscriptionDeletedError? {
        log:printInfo("onEventNotification invoked ", contentDistributionMessage = event);
        return {};
    }
}

listener websub:Listener subListener = new (9096);

SimpleWebsubService simpleSubscriberService = new;

SubscriberServiceConfiguration config = {
        target: "https://sample.topic.com/common/discovery",
        leaseSeconds: 36000,
        callback: "https://sample.subscriber.com/subscriber"
};

subListener.initialize(simpleSubscriberService, config, "subscriber");
```

Fixes [#1080](https://github.com/ballerina-platform/ballerina-standard-library/issues/1080)

### CODE REVIEW

**Review by shafreenAnfar — APPROVED** [review]

File: websub-ballerina/tests/websub_config_manual_attach_test.bal [hunk 1]
```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (hemikak) [inline websub-ballerina/tests/websub_config_manual_attach_test.bal:1]: Year should be 2021 ?

File: websub-ballerina/tests/websub_config_manual_attach_test.bal [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import ballerina/test;
+import ballerina/http;
+
+service class SimpleWebsubService {
+    *SubscriberService;
+    remote function onSubscriptionValidationDenied(SubscriptionDeniedError msg) returns Acknowledgement? {
+        log:printDebug("onSubscriptionValidationDenied invoked");
+        Acknowledgement ack = {
+                  headers: {"header1": "value"},
+                  body: {"formparam1": "value1"}
```
Comment (hemikak) [inline websub-ballerina/tests/websub_config_manual_attach_test.bal:27]: Indentation is wrong it seems.

File: websub-ballerina/tests/websub_config_manual_attach_test.bal (hunk 1 above)
Comment (ayeshLK) [reply websub-ballerina/tests/websub_config_manual_attach_test.bal:1]: Ack. Will fix this with next PR.

File: websub-ballerina/tests/websub_config_manual_attach_test.bal (hunk 2 above)
Comment (ayeshLK) [reply websub-ballerina/tests/websub_config_manual_attach_test.bal:27]: Ack. Will fix this with next PR.

File: websub-ballerina/sub_listener.bal [hunk 3]
```diff
@@ -65,13 +65,33 @@ public class Listener {
...
         }
     }
+
+    # Setup the provided Service with given configurations and attaches it to the listener
+    #
+    # + s - The `websub:SubscriberService` object to attach
+    # + configuration - `SubscriberServiceConfiguration` which should be incorporated into the provided Service 
+    # + name - The path of the Service to be hosted
+    # + return - An `error`, if an error occurred during the service attaching process
+    public function initialize(SubscriberService s, SubscriberServiceConfiguration configuration, string[]|string? name = ()) returns error? {
```
[the remark this answers is not in the document]
Comment (ayeshLK) [reply websub-ballerina/sub_listener.bal:81]: Ack. Will fix this with next PR.

**Pull request merged**